zookeeper-api/src/main/java/org/apache/helix/zookeeper/impl/client/DedicatedZkClient.java

```diff
@@ -372,6 +372,12 @@ public void asyncCreate(String path, Object datat, CreateMode mode,
     _rawZkClient.asyncCreate(path, datat, mode, cb);
   }
 
+  @Override
+  public void asyncCreate(String path, Object datat, CreateMode mode,
```

Comment:
It's better to support it for the dedicated zkclient only. For the other clients, especially customer-facing interface, I would prefer not to open this API for now.

Reply:
By doing so, I believe additional interface would help. Or you need to do the ugly default methods everywhere.

Reply:
There is trade-off: with additional interfaces, we need to maintain the additional interfaces, and we need to change the zkclient type with SessionAwareZkClient in HelixManager as well, because these BaseDataAccessors are using the zkclient in helix manager underneath. Let's discuss it further :)

Reply:
@jiajunwang PR updated.
We decided to keep current existing interfaces as they are. Instead, we use the srcSessionId in message as the expected session id to create message znode.